shimming-toolbox/shimmingtoolbox/masking/softmasks.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import nibabel as nib
+import numpy as np
+from scipy.ndimage import distance_transform_edt
+
+def create_softmask(fname_binmask, fname_softmask=None, type='2levels', soft_width=6, width_unit='mm', soft_value=0.5):
+    """
+    Create a soft mask from a binary mask by adding a soft zone around the binary mask.
+
+    Args:
+        fname_binmask (str): Path to the binary mask.
+        fname_softmask (str): Path to an existing soft mask. Used only if type is 'sum'.
+        type (str): Type of soft mask to create. Allowed types are: '2levels', 'linear', 'gaussian', 'sum'.
+        soft_width (float): Width of the soft zone.
+        soft_units (str): Units of the soft width ('mm' or 'px').
+        soft_value (float): Value of the intensity of the pixels in the soft zone. Used only if type is '2levels'.
+
+    Returns:
+        numpy.ndarray: 3D array containing the soft mask.
+    """
+    nifti_binmask = nib.load(fname_binmask)
+    binmask = nifti_binmask.get_fdata()
+    softmask = None
+    if fname_softmask is not None:
+        nifti_softmask = nib.load(fname_softmask)
+        softmask = nifti_softmask.get_fdata()
+
+    soft_width_px = convert_to_pixels(soft_width, width_unit, nifti_binmask.header)
+
+    if type == '2levels':
+        return create_two_levels_softmask(binmask, soft_width_px, soft_value)
+    elif type == 'linear':
+        return create_linear_softmask(binmask, soft_width_px)
+    elif type == 'gaussian':
+        return create_gaussian_softmask(binmask, soft_width_px)
+    if type == 'sum':
+        return add_softmask_to_binmask(binmask, softmask)
+    else:
+        raise ValueError("Invalid soft mask type. Must be one of: '2levels', 'linear', 'gaussian', 'sum'")
```

Comment:
Why was this removed?

```suggestion
  soft_maks_options = {
    '2levels': lambda: create_two_levels_softmask(binmask, soft_width_px, soft_value),
    'linear': lambda: create_linear_softmask(binmask, soft_width_px),
    'gaussian': lambda: create_gaussian_softmask(binmask, soft_width_px),
    'sum': lambda: add_softmask_to_binmask(binmask, softmask)
}

try:
    return soft_maks_options[type]()
except KeyError:
    raise ValueError(f"Invalid soft mask type. Must be one of: soft_maks_options.keys())
```

Reply:
Modifications applied